vcd/resource_vcd_external_network.go

```diff
@@ -0,0 +1,334 @@
+	log.Printf("[TRACE] external network read initiated")
+
+	vcdClient := meta.(*VCDClient)
+
+	externalNetworkRef, err := govcd.GetExternalNetworkByName(vcdClient.VCDClient, d.Id())
+	if err != nil || *externalNetworkRef == (types.ExternalNetworkReference{}) {
+		log.Printf("[DEBUG] Unable to find external network. Removing from tfstate")
+		d.SetId("")
+		return nil
+	}
+
+	externalNetwork := govcd.NewExternalNetwork(&vcdClient.VCDClient.Client)
+	externalNetwork.ExternalNetwork.HREF = externalNetworkRef.HREF
+	err = externalNetwork.Refresh()
+	if err != nil {
+		log.Printf("[DEBUG] Error fetching external network details %#v", err)
+		return fmt.Errorf("error fetching external network details %#v", err)
+	}
+
+	log.Printf("[TRACE] external network read completed: %#v", externalNetwork)
+	return nil
+}
+
+//update function for "delete_force", "delete_recursive" no actions needed
+func resourceVcdExternalNetworkUpdate(d *schema.ResourceData, m interface{}) error {
+	return nil
+}
+
+// Deletes a external network, optionally removing all objects in it as well
+func resourceVcdExternalNetworkDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] external network delete started")
+
+	vcdClient := meta.(*VCDClient)
+
+	externalNetworkRef, err := govcd.GetExternalNetworkByName(vcdClient.VCDClient, d.Id())
+	if err != nil || *externalNetworkRef == (types.ExternalNetworkReference{}) {
+		log.Printf("[DEBUG] Unable to find external network. Removing from tfstate")
+		d.SetId("")
+		return nil
+	}
+
+	externalNetwork := govcd.NewExternalNetwork(&vcdClient.VCDClient.Client)
+	externalNetwork.ExternalNetwork.HREF = externalNetworkRef.HREF
+	err = externalNetwork.Refresh()
+	if err != nil {
+		log.Printf("[DEBUG] Error fetching external network details %#v", err)
+		return fmt.Errorf("error fetching external network details %#v", err)
+	}
+
+	err = externalNetwork.DeleteWait()
+	if err != nil {
+		log.Printf("[DEBUG] Error removing external network %#v", err)
+		return fmt.Errorf("error removing external network %#v", err)
+	}
+
+	log.Printf("[TRACE] external network delete completed: %#v", externalNetwork)
+	return nil
+}
+
+// helper for tranforming the resource input into the ExternalNetwork structure
+// any cast operations or default values should be done here so that the create method is simple
+func getExternalNetworkInput(d *schema.ResourceData, vcdClient *VCDClient) (*types.ExternalNetwork, error) {
+	params := &types.ExternalNetwork{
+		Name:        d.Get("name").(string),
+		Xmlns:       "http://www.vmware.com/vcloud/extension/v1.5",
+		XmlnsVCloud: "http://www.vmware.com/vcloud/v1.5",
+		Configuration: &types.NetworkConfiguration{
+			Xmlns:                          "http://www.vmware.com/vcloud/v1.5",
+			FenceMode:                      d.Get("fence_mode").(string),
+			RetainNetInfoAcrossDeployments: d.Get("retain_net_info_across_deployments").(bool),
+		},
+		/*
```

Comment:
This commented out code is temporary? Missing feature which wasn't implemented yet?